### PR TITLE
chatterino7: init at 7.4.5

### PR DESCRIPTION
**Pull request by hauskens**
## Things done

Based on chatterino2 package using chatterino7 fork. Since binary name is identical it is not compatible with each other.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by nagy — CHANGES_REQUESTED** [review]
thanks, can you separate your addition to the maintainer list into its own commit? you can find good examples for the commit message in the history of that file. 

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

**Comment by hauskens** [comment]
will fix editorconfig and get the commits signed when later today

File: pkgs/applications/networking/instant-messengers/chatterino7/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  };
+  nativeBuildInputs = [ cmake pkg-config wrapQtAppsHook ];
+  buildInputs = [ qtbase qtsvg qtmultimedia qtimageformats qttools boost openssl libsecret ];
+
+  postInstall = lib.optionalString stdenv.isDarwin ''
+    mkdir -p "$out/Applications"
+    mv bin/chatterino.app "$out/Applications/"
+  '' + ''
+    mkdir -p $out/share/icons/hicolor/256x256/apps
+    cp $src/resources/icon.png $out/share/icons/hicolor/256x256/apps/chatterino.png
```
Comment (nagy) [inline pkgs/applications/networking/instant-messengers/chatterino7/default.nix:21]: Can you try this without `$src/` ? That should already be unpacked into the current directory and that makes working with nix-shell easier.

Comment (nagy) [reply pkgs/applications/networking/instant-messengers/chatterino7/default.nix]: no, you can just leave it as is then

**Comment by nagy** [comment]
can you remove the "v" prefix from your commit message and PR title?

**Comment by hauskens** [comment]
@nagy Done! Let me know if there is anything else i should look into

**Review by nagy — APPROVED** [review]

**Comment by wegank** [comment]
Looks like it doesn't build on darwin?

**Comment by hauskens** [comment]
I'm closing this one. I've gained more experience with Nix and i realize this is not really an optimal approach. Maybe i will re-visit this in the future.

**Pull request closed without merging**